### PR TITLE
fix(docs): fix rustdoc errors

### PR DESCRIPTION
**Pull request by StephenWakely**
A number of RustDoc errors have crept in. This fixes them. 

This also adds a CI job to test the rust doc.

One rust doc error has been left in for now to test this CI job.

### CODE REVIEW

File: src/amqp.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-//! Functionality supporting both the [sources::amqp] source and `[sinks::amqp]` sink.
+//! Functionality supporting both the `[crate::sources::amqp] source and `[crate::sinks::amqp]` sink.
```
Comment (spencergilbert) [inline src/amqp.rs:1]:
```suggestion
//! Functionality supporting both the `[crate::sources::amqp]` source and `[crate::sinks::amqp]` sink.
```

Missing the second backtick?

File: lib/vector-config-common/src/schema/gen.rs [hunk 2]
```diff
@@ -42,6 +42,7 @@ impl SchemaSettings {
 
     /// Creates a `Visitor` from the given closure and appends it to the list of
     /// [visitors](SchemaSettings::visitors) for these `SchemaSettings`.
+    #[allow(rustdoc::private_intra_doc_links)]
```
Comment (spencergilbert) [inline lib/vector-config-common/src/schema/gen.rs:45]: Are we not able to "fix" these?

File: src/amqp.rs (hunk 1 above)
Comment (spencergilbert) [reply src/amqp.rs:1]: or was this the intentionally broken one

File: lib/vector-config-common/src/schema/gen.rs (hunk 2 above)
Comment (StephenWakely) [reply lib/vector-config-common/src/schema/gen.rs:45]: I wasn't able to. The recommendation to pass `--document-private-items` did not fix the error, so I figured we could defer for now.

**Review by spencergilbert — APPROVED** [review]

**Pull request merged**